index.bs

```diff
@@ -1859,6 +1860,8 @@ The <dfn attribute for="XRInputSource">profiles</dfn> attribute is a [=/list=] o
 
 An <dfn for="XRInputSource">input profile name</dfn> is an ASCII lowercase {{DOMString}} containing no spaces, with separate words concatenated with a hyphen (`-`) character. A descriptive name should be chosen, using the prefered verbiage of the device vendor when possible. If the platform provides an appropriate identifier, such as a USB vendor and product ID, it MAY be used. Values that uniquely identify a single device, such as serial numbers, MUST NOT be used. The [=input profile name=] MUST NOT contain an indication of device handedness. If multiple user agents expose the same device, they SHOULD make an effort to report the same [=input profile name=]. The <a href="https://github.com/immersive-web/webxr-input-profiles/tree/master/packages/registry">WebXR Input Profiles Registry</a> is the recommended location for managing [=input profile name=]s.
 
+The <dfn attribute for="XRInputSource">alreadyVisible</dfn> attribute indicates that this input is visible and does not need to be rendered by the Website. Examples include the input source being between the user and the display or the input being shown by the operating system. 
```

Comment:
also hand input in AR?

Worth covering all common cases here

Reply:
In this particular case the developer could cover the physical input with pixels if they were drawing a large environment around the user in-spite of being in AR but I guess in general you wouldn't want to show the hands in AR and it is just hint to the developer which they can ignore.

Reply:
Yeah this is just a hint. Though the discussion does make me wonder if this should be an enum between "not-visible", "visible-real", and "visible-drawn". Probably overkill.

Reply:
Maybe: 
The <dfn attribute for="XRInputSource">skipRendering</dfn> attribute indicates that this input is visible and MAY NOT need to be rendered by the current session. If {{XRInputSource/skipRendering}} is <code>true</code>, the user agent MUST ensure that the [=XR input source=] is always visible.

Note: Examples of this are:
- the input source is between the user and the display 
- the input source is always rendered on top  by the operating system.

Reply:
> If {{XRInputSource/skipRendering}} is true, the user agent MUST ensure that the [=XR input source=] is always visible.

Hm... I don't know about that. I feel like there are examples of cases where the input source simply shouldn't be rendered at all where this would apply. For example: It feels natural to set this to `true` on `gaze` input sources? (You don't want to render anything in the user's head, after all.) We already tell devs to look at the  `targetRayMode` for this, but it would be nice to have this as a more universal signal.

Though that also brings up the question of rendering related graphics. For example: If I'm told to skip rendering a controller on Quest, should I still render any applicable pick rays? I assume the answer is yes, but that may be ambiguous given the proposed text. We should probably specify that this property is a hint about rendering the _physical input source_ only. Maybe include a non-normative note to say that things like pick rays and cursors should still be rendered if appropriate for the app and `targetRayMode`.

Reply:
> > If {{XRInputSource/skipRendering}} is true, the user agent MUST ensure that the [=XR input source=] is always visible.
> 
> Hm... I don't know about that. I feel like there are examples of cases where the input source simply shouldn't be rendered at all where this would apply. For example: It feels natural to set this to `true` on `gaze` input sources? (You don't want to render anything in the user's head, after all.) We already tell devs to look at the `targetRayMode` for this, but it would be nice to have this as a more universal signal.

Maybe:
`The {{XRInputSource/skipRendering}} attribute indicates that this input is visible and MAY NOT need to be rendered by the current session. If {{XRInputSource/skipRendering}} is true and the targetRayMode is "tracked-pointer", the user agent MUST ensure that a representation of the [=XR input source=] is always shown to the user.`

> Though that also brings up the question of rendering related graphics. For example: If I'm told to skip rendering a controller on Quest, should I still render any applicable pick rays? I assume the answer is yes, but that may be ambiguous given the proposed text. We should probably specify that this property is a hint about rendering the _physical input source_ only. Maybe include a non-normative note to say that things like pick rays and cursors should still be rendered if appropriate for the app and `targetRayMode`.

`NOTE: skipRendering is a hint about not rendering input sources such as controllers. Pick rays and cursor should still be rendered.`

Reply:
Both of those work for me! Maybe a minor tweak to the second one:

> NOTE: `skipRendering` is a hint that input sources such as controllers do not need to be rendered. Associated application-specific input visualizations such as pick rays and cursors should still be rendered.

Reply:
Thanks everyone for the feedback I will update it now.